.github/workflows/go-lint.yml

```diff
@@ -0,0 +1,29 @@
+name: Lint Go
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+
+env:
+  GOLANGCI_LINT_VERSION: v1.61.0
+jobs:
+  lint-go:
+    name: Lint Go
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: ./go.mod
+      - name: Run golangci-lint
```

Comment:
I would probably move this step into the ci workflow to run before build and test steps. If lint fails, good to exit early.  Maybe there's a reason you want them separated and I don't see it?

Would also simplify having one action/check to configure as required to be able to merge compared with two.

Reply:
In golangci-lint docs

> We recommend running this action in a job separate from other jobs (go test, etc.) because different jobs [run in parallel](https://help.github.com/en/actions/getting-started-with-github-actions/core-concepts-for-github-actions#job).

I guess we can merge the workflows and create separate job and confgiure build-test `needs: lint`